content/blog/migrating-a-cloud-application-to-kubernetes/index.md

```diff
@@ -0,0 +1,476 @@
+                        { name: "POSTGRES_PORT", value: String(5432) },
+                        { name: "DATABASE_NAME", value: postgresDatabase.name },
+                    ],
+                    resources: {
+                        limits: {
+                            memory: "500Mi",
+                            cpu: "500m"
+                        }
+                    }
+                }],
+            }
+        }
+    }}, {
+    provider: eksCluster.provider,
+});
+
+const serversideListener = new k8s.core.v1.Service("server-side-listener", {
+    metadata: { labels: serverAppLabels },
+    spec: {
+        type: "LoadBalancer",
+        ports: [{ port: 5000, targetPort: "http" }],
+        selector: serverAppLabels,
+    }}, {
+    provider: eksCluster.provider,
+    }
+);
+```
+
+And lastly, the client deployment.
+
+```typescript
+const clientAppName = "client";
+const clientAppLabels = { appClass: clientAppName };
+const clientDeployment = new k8s.apps.v1.Deployment("client-side-service", {
+    metadata: { labels: clientAppLabels },
+    spec: {
+        replicas: 2,
+        selector: { matchLabels: clientAppLabels },
+        template: {
+            metadata: { labels: clientAppLabels },
+            spec: {
+                containers: [{
+                    name: clientAppName,
+                    image: awsx.ecr.buildAndPushImage("client-side-service", "./clientside").image(),
+                    ports: [{ name: "http", containerPort: 3000 }],
+                    env: [
+                        { name: "SERVER_HOSTNAME", value: serversideListener.status.loadBalancer.ingress[0].hostname },
+                    ],
+                    resources: {
+                        limits: {
+                            memory: "500Mi",
+                            cpu: "500m"
+                        }
+                    },
+                }],
+            }
+        }
+    }}, {
+    provider: eksCluster.provider,
+});
+
+const clientsideListener = new k8s.core.v1.Service("client-side-listener", {
+    metadata: { labels: clientAppLabels },
+    spec: {
+        type: "LoadBalancer",
+        ports: [{ port: 3000, targetPort: "http" }],
+        selector: clientAppLabels,
+    }}, {
+    provider: eksCluster.provider,
+    }
+);
+```
+
+To make our Kubernetes application available on the Internet, we export the clientside listener's address. We can open a browser window with the URL and port to view our application.
+
+```typescript
+export const URL = clientsideListener.status.loadBalancer.ingress[0].hostname;
+```
+
+In this example, we went over the benefits that Kubernetes offers, and showed how to use Pulumi to convert an application to Kubernetes and deploy it on AWS. Although it might seem uneccessary to use such a system for only a single voting app, the advantages of running your project in the form of containerized services quickly become apprant as your application grows in size.  
```

Comment:
```suggestion
In this example, we went over the benefits that Kubernetes offers and showed how to use Pulumi to convert an application to Kubernetes and deploy it on AWS. Although it might seem unnecessary to use such a system for only a single voting app, the advantages of running your project in the form of containerized services quickly become apparent as your application grows in size.  
```